src/System.IO.Ports/src/System/IO/Ports/SafeSerialDeviceHandle.Unix.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Net.Sockets;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Ports
+{
+    internal sealed class SafeSerialDeviceHandle : SafeHandle
+    {
+        private SafeSerialDeviceHandle() : base(new IntPtr(-1), ownsHandle: true)
+        {
+        }
+
+        internal static SafeSerialDeviceHandle Open(string portName)
+        {
+            Debug.Assert(portName != null);
+            SafeSerialDeviceHandle handle = Interop.Serial.SerialPortOpen(portName);
+
+            if (handle.IsInvalid)
+            {
+                // exception type is matching Windows
+                throw new UnauthorizedAccessException(
+                    string.Format(SR.UnauthorizedAccess_IODenied_Port, portName),
```

Comment:
I believe this should be `SR.Format` not `string.Format` because it produces a better result in AoT where exception messages may be stripped. See SR.Format use elsewhere.